pkgs/applications/virtualization/singularity/default.nix

```diff
@@ -1,62 +1,79 @@
-{ stdenv
+{stdenv
+, removeReferencesTo
+, lib
+, fetchgit
 , fetchFromGitHub
-, autoreconfHook
-, gnutar
-, which
-, gnugrep
+, utillinux
+, openssl
 , coreutils
-, python
-, e2fsprogs
+, gawk
+, go
+, which
 , makeWrapper
 , squashfsTools
-, gzip
-, gnused
-, curl
-, utillinux
-, libarchive
-, file
- }:
+, buildGoPackage}:
+
+with lib;
 
-stdenv.mkDerivation rec {
+buildGoPackage rec {
   name = "singularity-${version}";
-  version = "2.6.0";
+  version = "3.0.1";
+
+  src = fetchFromGitHub {
+    owner = "sylabs";
+    repo = "singularity";
+    rev = "v${version}";
+    sha256 = "1wpsd0il2ipa2n5cnbj8dzs095jycdryq2rx62kikbq7ahzz4fsi";
+  };
+
+  goPackagePath = "github.com/sylabs/singularity";
+  goDeps = ./deps.nix;
+
+  buildInputs = [ go removeReferencesTo utillinux openssl which makeWrapper ];
```

Comment:
```suggestion
  buildInputs = [ openssl ];
  nativeBuildInputs = [ makeWrapper which utillinux ];
```

Reply:
utillinux and which are only used at runtime, right?

Reply:
Yes, I have patched and tested.